dd-sdk-android/src/test/kotlin/com/datadog/android/DatadogInterceptorWithoutTracesTest.kt

```diff
@@ -0,0 +1,319 @@
+import okhttp3.Protocol
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.Response
+import okhttp3.ResponseBody
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class DatadogInterceptorWithoutTracesTest {
+
+    lateinit var testedInterceptor: TracingInterceptor
+
+    // region Mocks
+
+    @Mock
+    lateinit var mockTracer: Tracer
+
+    @Mock
+    lateinit var mockLocalTracer: Tracer
+
+    @Mock
+    lateinit var mockSpanBuilder: Tracer.SpanBuilder
+
+    @Mock
+    lateinit var mockSpanContext: DDSpanContext
+
+    @Mock(extraInterfaces = [MutableSpan::class])
+    lateinit var mockSpan: Span
+
+    @Mock
+    lateinit var mockChain: Interceptor.Chain
+
+    @Mock
+    lateinit var mockRequestListener: TracedRequestListener
+
+    lateinit var mockDevLogHandler: LogHandler
+
+    lateinit var mockAppContext: Context
+
+    @Mock
+    lateinit var mockRumMonitor: RumMonitor
+
+    // endregion
+
+    // region Fakes
+
+    @RegexForgery(TracingInterceptorTest.HOSTNAME_PATTERN)
+    lateinit var fakeHostName: String
+
+    @RegexForgery(TracingInterceptorTest.IPV4_PATTERN)
+    lateinit var fakeHostIp: String
+
+    lateinit var fakeMethod: String
+    var fakeBody: String? = null
+    var fakeMediaType: MediaType? = null
+
+    @StringForgery(StringForgeryType.ASCII)
+    lateinit var fakeResponseBody: String
+
+    lateinit var fakeUrl: String
+
+    @Forgery
+    lateinit var fakeConfig: DatadogConfig.FeatureConfig
+
+    @StringForgery(StringForgeryType.ALPHABETICAL)
+    lateinit var fakePackageName: String
+
+    @RegexForgery("\\d(\\.\\d){3}")
+    lateinit var fakePackageVersion: String
+
+    lateinit var fakeRequest: Request
+    lateinit var fakeResponse: Response
+
+    @StringForgery(StringForgeryType.HEXADECIMAL)
+    lateinit var fakeSpanId: String
+
+    @StringForgery(StringForgeryType.HEXADECIMAL)
+    lateinit var fakeTraceId: String
+
+    // endregion
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        mockDevLogHandler = mockDevLogHandler()
+        mockAppContext = mockContext(fakePackageName, fakePackageVersion)
+        Datadog.setVerbosity(Log.VERBOSE)
+
+        whenever(mockTracer.buildSpan(TracingInterceptor.SPAN_NAME)) doReturn mockSpanBuilder
+        whenever(mockSpanBuilder.asChildOf(null as SpanContext?)) doReturn mockSpanBuilder
+        whenever(mockSpanBuilder.start()) doReturn mockSpan
```

Comment:
I am not sure you need these as the span will never be created in this case.

Reply:
Indeed.